foam_internalwrapper/foam_mesh.py

```diff
@@ -0,0 +1,936 @@
+
+        """
+
+        return False
+
+    def has_faces(self):
+        """ Check if the mesh has faces
+
+        Returns
+        -------
+        bool
+            True of there are faces inside the mesh,
+            False otherwise
+
+        """
+        numberFaces = foamface.getFaceCount(self.name)
+        return numberFaces > 0
+
+    def has_cells(self):
+        """ Check if the mesh has cells
+
+        Returns
+        -------
+        bool
+            True of there are cells inside the mesh,
+            False otherwise
+
+        """
+        numberCells = foamface.getCellCount(self.name)
+        return numberCells > 0
+
+    def write(self):
+        """ Writes mesh from OpenFOAM's objectRegistry to disk
+
+        """
+
+        foamface.writeMesh(self.name)
+
+    def write_data(self):
+        """ Writes mesh related data from OpenFOAM's objectRegistry to disk
+
+        """
+
+        # currently only cell data is written
+        dataNames = foamface.getCellDataNames(self.name)
+        for dataName in dataNames:
+            if dataName != '':
+                foamface.writeCellData(self.name, dataName)
+        dataNames = foamface.getCellVectorDataNames(self.name)
+        for dataName in dataNames:
+            if dataName != '':
+                foamface.writeCellVectorData(self.name, dataName)
+
+    def update_data(self):
+        """ Updates mesh related data from disk to OpenFOAM's objectRegistry
+
+        """
+
+        # currently only cell data is updated
+        dataNames = foamface.getCellDataNames(self.name)
+        for dataName in dataNames:
+            foamface.updateCellData(self.name, dataName)
+        dataNames = foamface.getCellVectorDataNames(self.name)
+        for dataName in dataNames:
+            foamface.updateCellVectorData(self.name, dataName)
+
+    def generate_uuidmapping(self, nPoints, nEdges, nFaces, nCells):
+        '''generate uuid mapping assuming continuous ordering of mesh objects
+
+        '''
+
+        for point in range(nPoints):
+            uuid = self._generate_uuid()
+            self._uuidToFoamLabel[uuid] = point
+            self._foamPointLabelToUuid[point] = uuid
+
+        for edge in range(nEdges):
+            uuid = self._generate_uuid()
+            self._uuidToFoamLabel[uuid] = edge
+            self._foamEdgeLabelToUuid[edge] = uuid
+
+        for face in range(nFaces):
+            uuid = self._generate_uuid()
+            self._uuidToFoamLabel[uuid] = face
+            self._foamFaceLabelToUuid[face] = uuid
+
+        for cell in range(nCells):
+            uuid = self._generate_uuid()
+            self._uuidToFoamLabel[uuid] = cell
+            self._foamCellLabelToUuid[cell] = uuid
+
+    def _generate_uuid(self):
+        """ Provides an uuid for the object
+
+        Provides san uuid as defined in the standard RFC 4122
+        """
+
+        return uuid.uuid4()
+
+    def modifyNumerics(self, SP):
```

Comment:
Every method which is not part of ABCMesh was redefined as utility function. 
However, we left the method getXYZUVW() in the current commit because we plan to replace it when we incorporate the native post-processing of simphony.